src/event.ts

```diff
@@ -61,40 +60,61 @@ export default class EventManager {
     userId: string,
     payload: Partial<TokenomicsEvent>
   ) {
+    const data = {
+      appId,
+      eventId,
+      userId,
+      payload,
+      timestamp: Date.now(),
+    };
+    const signature = this.ain.wallet.sign(stringify(data));
     return axios
-      .put(`${this.baseUrl}/update`, {
-        appId,
-        eventId,
-        userId,
-        payload,
-        accessAinAddress: this.accessAddress,
-        signature: this.signature,
-        data: this.signatureData,
-      })
+      .put(`${this.baseUrl}/update`, { data, signature })
       .then((res) => res.data)
       .catch((e) => {
         throw e.response.data;
       });
   }
 
   deleteEvent(appId: string, eventId: string, userId: string) {
+    const data = { appId, eventId, userId, timestamp: Date.now() };
+    const signature = this.ain.wallet.sign(stringify(data));
     return axios
-      .delete(`${this.baseUrl}/delete`, {
-        data: {
-          appId,
-          eventId,
-          userId,
-          accessAinAddress: this.accessAddress,
-          signature: this.signature,
-          data: this.signatureData,
-        },
+      .delete(`${this.baseUrl}/delete`, { 
+        data: { data, signature },
       })
       .then((res) => res.data)
       .catch((e) => {
         throw e.response.data;
       });
   }
 
+  addEventActivity({
+    appId,
+    userId,
+    eventId,
+    smartGalleryPosId,
+    taskInstanceId,
+    data: _data,
+  }: AddEventActivityParams) {
```

Comment:
How about adding a return type? When I used SDK, It was very helpful.  
(It is not declared in all functions of this SDK, but it would be good to add them gradually.)

Reply:
great idea! done.
related updates on the server side: https://bitbucket.org/comcomai/nft-server/pull-requests/33/feature-liayoo-return-activity-id